### PR TITLE
Remove unnecessary CSS class `word-count` from information panel

### PR DESCRIPTION
**Pull request by david-szabo97**
## Description
We already have a CSS class to style the count: `.table-of-contents__number`
This PR removes the unnecessary `.word-count` class.

## Types of changes
Code Quality

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by Copons** [comment]
I think this is risky.
We are saving one duplicated CSS rule at the expense of changing the rendered content of a publicly exported component.

While it's true that `WordCount` is only used once inside Gutenberg, and your change addresses that, we might end up accidentally breaking external consumers of the component.

File: packages/editor/src/components/word-count/index.js [hunk 1]
```diff
@@ -13,11 +13,7 @@ function WordCount( { content } ) {
 	 */
 	const wordCountType = _x( 'words', 'Word count type. Do not translate!' );
 
-	return (
-		<span className="word-count">
```
Comment (vindl) [inline packages/editor/src/components/word-count/index.js:17]: Would it make sense to retain span wrapper here and only change the classname?

Comment (david-szabo97) [reply packages/editor/src/components/word-count/index.js:17]: I'd keep it decoupled. Adding the `.table-of-contents__number` class in there would make it tightly coupled with `table-of-contents`.

**Comment by Copons** [comment]
@david-szabo97 If you think the additional specificity is too much of a burden, what about leaving the `WordCount` render as it is right now (with the whole `span.word-count`) to keep it backward compatible, and instead adding a `span.table-of-contents__number` in the Table of Contents to "mirror" the other list items?

E.g.
```jsx
// word-count/index.js
<span className="word-count">
	{ wordCount( content, wordCountType ) }
</span>

// table-of-contents/panel.js
<li className="table-of-contents__count">
	{ __( 'Words' ) }
	<span className="table-of-contents__number">
		<WordCount />
	</span>
</li>
```
```scss
// table-of-contents/style.scss
.table-of-contents__number {
	font-size: 21px;
	font-weight: 400;
	line-height: 30px;
	color: $dark-gray-500;
}
```

**Comment by david-szabo97** [comment]
Closing this. Better to keep it as-is to avoid changing public-facing content.

**Pull request closed without merging**